services/code_hosting/git_hub_service.rb

```diff
@@ -1,131 +1,409 @@
+      end
+
+      # Does the client with the associated credentials have access to the specified repo?
+      # @return [Bool]
+      def access_to_repo?(provider_credential: nil, repo_url: nil)
+        self.client(provider_credential.api_token).repository?(repo_from_url(repo_url))
+      end
+
+      # @return [Array<String>] names of the branches for the given repo
+      def branch_names(provider_credential: nil, repo_full_name: nil)
+        self.client(provider_credential.api_token).branches(repo_full_name).map(&:name)
+      end
+
+      def repo_from_url(repo_url)
+        return repo_url.sub("https://github.com/", "")
+      end
+
+      def url_from_repo(repo)
+        return repo.include?("https://github.com/") ? repo : "https://github.com/" + repo
+      end
+    end
 
     # The email is actually optional for API access
     # However we ask for the email on login, as we also plan on doing commits for the user
     # and this way we can make sure to configure things properly for git to use the email
     attr_accessor :provider_credential
 
-    def initialize(provider_credential: nil)
-      self.provider_credential = provider_credential
+    # @return [Project]
+    attr_accessor :project
 
+    def initialize(provider_credential: nil, project: nil)
+      self.provider_credential = provider_credential
+      raise "Project instance not provided or wrong type parameter" if project.nil? || !project&.kind_of?(Project)
+      self.project = project
       @_client = Octokit::Client.new(access_token: provider_credential.api_token)
       Octokit.auto_paginate = true # TODO: just for now, we probably should do smart pagination in the future
     end
 
+    def status_context
+      "fastlane.ci: " + @project.id
+    end
+
     def client
       @_client
     end
 
+    # @return [Bool] whether the session is valid or not.
     def session_valid?
-      client.login.to_s.length > 0
+      self.client.login.to_s.length > 0
     rescue StandardError
       false
     end
 
     def username
-      client.login
+      self.client.login
     end
 
-    # returns all open pull requests on given repo
-    # branches should be nil if you want all branches to be considered
-    def open_pull_requests(repo_full_name: nil, branches: nil)
-      all_open_pull_requests = client.pull_requests(repo_full_name, state: "open")
+    # Returns the urls of the pull requests for a given branch and state.
+    # If branches is not provided, all target branches are considered.
+    # @param [Array<String>, nil] branches, Either an array of target branches names or nil.
+    # @param [String] state, Either open, closed, or all to filter by state. Default: open.
+    # @return [String] HTML URL for the given pull request query.
+    def pull_requests(branches: nil, state: "open")
+      all_open_pull_requests = client.pull_requests(self.project.repo_config.full_name, state: state)
```

Comment:
variable name is wrong now, could open or closed

Reply:
True, renaming incoming.